client/signup/config/test/index.js

```diff
@@ -17,4 +19,57 @@ describe( 'index', () => {
 			);
 		}
 	} );
+
+	test( 'All steps should have the same dependencies provided', () => {
+		const stepModuleMap = getStepModuleMap();
+		const moduleStepMap = {};
+		Object.entries( stepModuleMap ).forEach( ( [ step, module ] ) => {
+			if ( ! moduleStepMap[ module ] ) {
+				moduleStepMap[ module ] = [];
+			}
+			moduleStepMap[ module ].push( step );
+		} );
+		const allStepDefinitions = generateSteps();
+
+		Object.values( moduleStepMap ).forEach( ( moduleRelatedSteps ) => {
+			const allDependenciesProvidedForModule = new Set();
+			if ( moduleRelatedSteps.length > 1 && allStepDefinitions ) {
+				moduleRelatedSteps.forEach( ( step ) => {
+					( allStepDefinitions[ step ]?.providesDependencies ?? [] ).forEach( ( d ) =>
+						allDependenciesProvidedForModule.add( d )
+					);
+				} );
+				const allDependenciesSortedArray = Array.from( allDependenciesProvidedForModule );
+				allDependenciesSortedArray.sort();
+
+				moduleRelatedSteps.forEach( ( step ) => {
+					if ( allStepDefinitions[ step ] ) {
+						const stepProvidedDependencies = (
+							allStepDefinitions[ step ]?.providesDependencies ?? []
+						).slice();
```

Comment:
I guess if we don't sort, we won't need to clone the array either?

```suggestion
						const stepProvidedDependencies = allStepDefinitions[ step ]?.providesDependencies ?? [];
```